readthedocs/projects/views/private.py

```diff
@@ -347,7 +349,33 @@ def get_form_kwargs(self, step=None):
 
     def get_template_names(self):
         """Return template names based on step name."""
-        return "projects/import_{}.html".format(self.steps.current)
+        return f"projects/import_{self.steps.current}.html"
+
+    def process_step(self, form):
+        if isinstance(form, ProjectBasicsForm):
+            remote_repository = form.cleaned_data.get("remote_repository")
+            if remote_repository and remote_repository.vcs_provider == GITHUB:
+                for yaml in [
+                    ".readthedocs.yaml",
+                    ".readthedocs.yml",
+                    "readthedocs.yaml",
+                    "readthedocs.yml",
+                ]:
+                    try:
+                        response = requests.head(
+                            f"https://api.github.com/repos/{remote_repository.full_name}/contents/{yaml}",
+                            timeout=1,
+                        )
```

Comment:
We should probably use the oauth client from the connected remote accounts first. Otherwise, this won't work for private projects.

Reply:
Done in 69e426272